src/Common/ThreadPool.cpp

```diff
@@ -379,82 +391,59 @@ void ThreadPoolImpl<Thread>::worker(typename std::list<Thread>::iterator thread_
             job = std::move(const_cast<Job &>(jobs.top().job));
             parent_thead_trace_context = std::move(const_cast<DB::OpenTelemetry::TracingContextOnThread &>(jobs.top().thread_trace_context));
             jobs.pop();
+
+            /// We don't run jobs after `shutdown` is set, but we have to properly dequeue all jobs and finish them.
+            if (shutdown)
+            {
+                parent_thead_trace_context.reset();
```

Comment:
Can we put this into `SCOPE_EXIT`?

Reply:
Yes, good point

Reply:
Oh, actually reset() is useless, because it is done on a copy of actual context. And this looks like an error. WDYT? (I'm not closely familiar with our tracing implementation yet)

Reply:
It was introduced with the initial trace context propagation commit here https://github.com/ClickHouse/ClickHouse/pull/39010/files#diff-58357c9febd25c282bf2d3ba4fc0ae7e6f134049f3d95ec7566058fc341c4057R308
And because we moved "job finish" motivation does not work any more - it will be destructed immediately on job done. So my removal is for good.